vectorai/write.py

```diff
@@ -687,3 +687,34 @@ def edit_documents(self, collection_name: str, edits: Dict, workers: int = 1):
             "failed": len(failed),
             "failed_document_ids": failed,
         }
+
+    def retrieve_and_encode(
+        self, 
+        collection_name: str,
+        models: Dict[str, Callable] = {},
+        chunksize: int = 15,
+        workers: int = 1,
+        verbose: bool=False,
+        use_bulk_encode: bool=False,
+        overwrite: bool=True,
+        show_progress_bar: bool=True):
+        docs = self.retrieve_documents(collection_name, page_size=chunksize)
+        docs['cursor'] = None
+        failed_all = {
+            "inserted_successfully": 0,
+            "failed": 0,
+            "failed_document_ids": [] 
+        }
+
+        while len(docs['documents']) > 0:
+            docs = self.retrieve_documents(collection_name, cursor=docs['cursor'],
```

Comment:
you should be using retrieve  documents of only specific fields and then bulk edit instead.